lib/matplotlib/axes/_base.py

```diff
@@ -176,7 +181,7 @@ def __call__(self, *args, **kwargs):
 
     def set_lineprops(self, line, **kwargs):
         assert self.command == 'plot', 'set_lineprops only works with "plot"'
-        for key, val in six.iteritems(kwargs):
+        for key, val in _sorted_iteritems(kwargs):
```

Comment:
This is looping over calling `set_*` methods which only mutates Artists, not adds them.